stateful/runme
# Refactor parsing and expose new WASM interface

Highlights:
* Parsing is simplified and refactored to handle edge cases more easily like nested fenced code blocks.
* Rendering process exposes each block separately so that when using the Notebook interface, we have blocks corresponding to cells.

To test the new WASM interface, run:

```
make wasm
cd examples/web
python -m http.server 9000
```

And open the address in a browser. You should see the new output. Check out `examples/web/index.html` to learn more about the new global functions.

## Todo

- [x] Go through these MDs:
   * [DevOps Handbook](https://github.com/freeCodeCamp/freeCodeCamp/blob/084bf7600f2f8fcdcb2bb82abe2f842d457b270a/docs/devops.md)
- [x] Implement the case when editing a block results in its split. For example, a block containing a single paragraph might be edited and converted into two paragraphs which result in two blocks each containing a single paragraph.

Closes https://github.com/stateful/runme/issues/49 https://github.com/stateful/runme/issues/50
